File: backend/models/wifi.py
from pydantic import BaseModel


class WifiConfig(BaseModel):
    Enable: bool | None = None
    Status: bool | None = None
    ConfigurationMode: bool | None = None
    TriggerAutoChannelSelection: bool | None = None


class WifiEnableRequest(BaseModel):
    value: bool
    temporary: bool | None = None
    source: str | None = None


class WifiStatusRequest(BaseModel):
    Status: bool


class WifiPairingRequest(BaseModel):
    clientPIN: str


class GuestConfig(BaseModel):
    Enable: bool

File: frontend/src/app/models/wifi.ts
export interface WifiConfig {
  Enable?: boolean;
  Status?: boolean;
  ConfigurationMode?: boolean;
  TriggerAutoChannelSelection?: boolean;
}

export interface WifiEnableRequest {
  value: boolean;
  temporary?: boolean;
  source?: string;
}

export interface WifiStatusRequest {
  status: boolean;
}

export interface WifiPairingRequest {
  clientPin: string;
}

export interface GuestConfig {
  enable: boolean;
}
